Update visual design and compatibility for SonarQube 10.8 / Community Build 24.12

fourls opened the pull request:
This PR updates DelphiLint for SonarQube 10.8 / Community Build 24.12 support, including a few relatively major changes:

* Client-side resources (e.g. images) are now automatically collected and built, rather than manually added to each project
* Rule descriptions no longer use the monolithic `htmlDesc` field, which is removed in 10.8 / 24.12 - they are now split into description sections, which are displayed in a tabbed view
* The taxonomy mode (MQR vs. Standard Experience) in 10.8 / 24.12 is now supported
* Visual styling has been updated to match SonarQube 10.8

Some images:

![image](https://github.com/user-attachments/assets/5daecb79-f674-475c-b5c3-91ab961e17e6)

![image](https://github.com/user-attachments/assets/1e45baa2-2aca-4504-b1a2-e1fb7d99d0b2)

![image](https://github.com/user-attachments/assets/0c0ea15c-b65e-4a3c-b7b2-0c92c3a04d80)

![image](https://github.com/user-attachments/assets/fb8f5549-468a-4b60-97bf-711d5052257f)

fourls commented:
The format check failure is due to an issue with the upstream action - #79 fixes the problem.

Cirras commented on server/delphilint-server/src/main/java/au/com/integradev/delphilint/remote/ImpactSeverity.java:
```diff
@@ -22,7 +22,9 @@
   MEDIUM,
-  HIGH;
+  HIGH,
+  INFO,
```
Should `INFO` be the 1st or the 4th value? (and in corresponding case statements too)

fourls replied:
Probably the first. I originally put it as the 4th everywhere, but that caused ordering issues when determining the maximum severity of an issue clientside.

fourls replied:
Done.

Cirras commented on server/delphilint-server/src/main/java/au/com/integradev/delphilint/remote/sonarqube/SonarQubeHost.java:
```diff
@@ -191,6 +192,34 @@ public Map<String, String> getRuleNamesByRuleKey() throws SonarHostException {
     return ruleMap;
   }
 
+  private boolean getIsTaxonomyCleanCode() throws SonarHostException {
```
Might be overkill, but should this just be `getTaxonomy` with a taxonomy enum?

fourls replied:
I guess I can make it an enum. It just seemed a bit unnecessary considering it's a single totally internal routine whose routine value is only used as a conditional check.

fourls replied:
Done.

Cirras reviewed (changes requested):
Great work! A couple of minor questions, and I'm yet to actually test - but code changes are looking good overall.

fourls commented:
Rebased on top of #81

Cirras reviewed (changes requested):
Noticed this bug in testing.
![image](https://github.com/user-attachments/assets/bd2078b4-91c1-4fde-b723-433f53f97e65)

Also, what are our feelings about the appearance of the new tabs in light mode?
I'm so-so on the darkened-background with thick black line at the bottom.
![image](https://github.com/user-attachments/assets/80138bea-0a0f-4531-9dcd-1cc8860c5316)

fourls commented:
Fixed the bug and updated the styling of the tabs: 

![image](https://github.com/user-attachments/assets/201dcc46-ecc5-48db-9e4f-a09d6c513baa)

![image](https://github.com/user-attachments/assets/822b9a52-e3d6-44a9-a071-4c45785ba861)

The selected tab is `clBtnFace` and the hovered tab is `clBtnShadow`.

Cirras reviewed (changes requested):
Overall looks much better - I have some thoughts on further tweaks.

* Should we could remove the top line from the tab bar?
* Was there a reason we couldn't use the same blue highlights from the issue list?

Here's a mockup of the top line being removed
![image](https://github.com/user-attachments/assets/1f4cb191-1bc4-413a-801b-1fcac0ba9881)


Same as before, but the blue highlights being used instead:
![image](https://github.com/user-attachments/assets/692c2993-260b-4143-86b3-749b36375763)


Here's a mockup where we've gone back to using a fat line at the bottom for our "selected" indicator:
![image](https://github.com/user-attachments/assets/4f5c19e5-d6b1-4626-ba53-9a50c2b5ff80)

fourls commented:
Top line removed.

> Was there a reason we couldn't use the same blue highlights from the issue list?

It's not clear which (if any) system colour is being used for these highlights. As a concession I've changed the button colours to be a little brighter:

![image](https://github.com/user-attachments/assets/76343adb-bfcc-44b6-a3c7-2875a0d05d0d)
![image](https://github.com/user-attachments/assets/76c09661-bc2b-4fb1-9bfa-7b9db473f500)

Cirras reviewed (approved):
Looks good to me! 👍
